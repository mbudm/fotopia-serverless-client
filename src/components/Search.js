import React from 'react';

const Search = () => (
  <div>
    <h2>Search</h2>
  </div>
);

export default Search;
