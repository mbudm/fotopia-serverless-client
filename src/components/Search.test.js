// Link.react.test.js
import React from 'react';
import { Search } from './Search';
import renderer from 'react-test-renderer';

test('Search with no result array', () => {
  const component = renderer.create(
    <Search />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Search with 0 results', () => {
  const component = renderer.create(
    <Search results={[]}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Search with 1 result', () => {
  const component = renderer.create(
    <Search results={[{
      id: 'image_guid',
      img_location: 'myImg.jpg',
      img_thumb_location: 'myImg-thumb.jpg'
    }]}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Search with multiple results', () => {
  const component = renderer.create(
    <Search results={[{
      id: 'image_guid',
      img_location: 'myImg.jpg',
      img_thumb_location: 'myImg-thumb.jpg'
    },
    {
      id: 'image_2_guid',
      img_location: 'myImg2.jpg',
      img_thumb_location: 'myImg2-thumb.jpg'
    }]}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
