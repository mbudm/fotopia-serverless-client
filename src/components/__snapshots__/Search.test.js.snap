// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search with 0 results 1`] = `
<div>
  <h2>
    Search
  </h2>
  <button
    onClick={undefined}
  >
    Get latest photos
  </button>
  <section
    className="row"
  />
</div>
`;

exports[`Search with 1 result 1`] = `
<div>
  <h2>
    Search
  </h2>
  <button
    onClick={undefined}
  >
    Get latest photos
  </button>
  <section
    className="row"
  >
    <figure
      className="col-md-2 center-block"
    >
      <img
        alt=""
        className="img-responsive"
        src="myImg.jpg"
      />
    </figure>
  </section>
</div>
`;

exports[`Search with multiple results 1`] = `
<div>
  <h2>
    Search
  </h2>
  <button
    onClick={undefined}
  >
    Get latest photos
  </button>
  <section
    className="row"
  >
    <figure
      className="col-md-2 center-block"
    >
      <img
        alt=""
        className="img-responsive"
        src="myImg.jpg"
      />
    </figure>
    <figure
      className="col-md-2 center-block"
    >
      <img
        alt=""
        className="img-responsive"
        src="myImg2.jpg"
      />
    </figure>
  </section>
</div>
`;

exports[`Search with no result array 1`] = `
<div>
  <h2>
    Search
  </h2>
  <button
    onClick={undefined}
  >
    Get latest photos
  </button>
  <img
    alt="searching"
    src="loader.svg"
  />
</div>
`;
